### ISSUE TITLE
Obsoleted parameterless constructors can be used as generic arguments with no warnings

### ISSUE THREAD

**Issue by Arthri** [dotnet/roslyn#73519]
**Version Used**: 4.8.0-7.23558.1 (e0917286)

**Steps to Reproduce**:
Run the following code([sharplab](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAsgHgFoD4AUBKA3ALABQJAAgCwAEaAKjrlQO4AWApgE5tW1UhUA7NkzxUA3gF8S5AMxUyAJiroSYklQ1UA2mQCMAOgDyAIwDOAewhtgbbACI7UKsA4BXNrgC66zWTnpRSRIpYiA=))
```cs
M<Z>();

void M<T>() where T : new() {}

public class Z
{
    [System.Obsolete("", true)]
    public Z() {}
}
```

**Diagnostic Id**: `CS0619: 'Z.Z()' is obsolete: ''`

**Expected Behavior**: An error because the obsoleted parameterless constructor was used indirectly

**Actual Behavior**: No warnings or errors

**Comment by CyrusNajmabadi** [comment]
We could consider this as a warning wave concern.
